### PR TITLE
Add support for Flatcar Container Linux

### PR DESCRIPTION
**Pull request by krnowak**
## Description

This adds support Flatcar Container Linux, which is a fork of CoreOS Container Linux, and is compatible with it. We currently [carry a patch in our wa-linux-agent package](https://github.com/flatcar-linux/coreos-overlay/blob/main/app-emulation/wa-linux-agent/files/0001-Support-flatcar.patch), and now I'm trying to upstream it.

I wasn't sure if I also need to update the code in `bin/waagent2.0`, which seems to be an older version of the project, so for now I left it alone. Please let me know if it needs updating too.

---

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [x] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def get_data_files(name, version, fullname):  # pylint: disable=R0912
         set_udev_files(data_files)
         set_systemd_files(data_files, dest=systemd_dir_path,
                           src=["init/arch/waagent.service"])
-    elif name == 'coreos':
+    elif name == 'coreos' or name == 'flatcar':  # pylint: disable=R1714
```
Comment (larohra) [inline setup.py:126]: Can you please resolve this pylint error - 

> # pylint: disable=R1714

File: azurelinuxagent/common/osutil/factory.py [hunk 2]
```diff
@@ -81,7 +81,8 @@ def _get_osutil(distro_name, distro_code_name, distro_version, distro_full_name)
     if distro_name == "kali":
         return DebianOSBaseUtil()
 
-    if distro_name == "coreos" or distro_code_name == "coreos":
+    if distro_name == "flatcar" or distro_code_name == "flatcar" \
```
Comment (larohra) [inline azurelinuxagent/common/osutil/factory.py:84]: NIT: Consider using `in` for better readability

Eg: 
```
if distro_name in ("flatcar", "coreos") or distro_code_name in ("flatcar", "coreos")
```

File: azurelinuxagent/pa/deprovision/factory.py [hunk 3]
```diff
@@ -35,7 +35,7 @@ def get_deprovision_handler(distro_name=DISTRO_NAME,
             return Ubuntu1804DeprovisionHandler()
         else:
             return UbuntuDeprovisionHandler()
-    if distro_name == "coreos":
+    if distro_name == "flatcar" or distro_name == "coreos":
```
Comment (larohra) [inline azurelinuxagent/pa/deprovision/factory.py:38]: same as above, consider using `in`

**Review by larohra — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (krnowak) [reply setup.py]: Done.

File: azurelinuxagent/common/osutil/factory.py (hunk 2 above)
Comment (krnowak) [reply azurelinuxagent/common/osutil/factory.py]: Done.

File: azurelinuxagent/pa/deprovision/factory.py (hunk 3 above)
Comment (krnowak) [reply azurelinuxagent/pa/deprovision/factory.py]: Done.

**Comment by krnowak** [comment]
Updated.

**Review by larohra — APPROVED** [review]
Looks good to me, running test automation suite

**Comment by larohra** [comment]
DCR ok - https://tuxgold.corp.microsoft.com/job/AzLinux/job/DungeonCrawler/job/DungeonCrawler.larohra3/217/artifact/results/

**Review by kevinclark19a — APPROVED** [review]

**Pull request merged**